core-services/src/main/java/org/zalando/nakadi/service/SchemaService.java

```diff
@@ -324,6 +326,22 @@ public org.apache.avro.Schema getAvroSchema(final String etName,
         });
     }
 
+    @Override
+    public String getAvroSchemaVersion(final String name, final org.apache.avro.Schema schema)
+            throws NoSuchSchemaException {
+        final NameSchema key = new NameSchema(name, schema);
+        return schemaVersionCache.computeIfAbsent(key, (nameSchema) ->
```

Comment:
Note for myself : review this again.

Reply:
This is not the best approach, but it will work for now, and we should improve this later. 
One easy optimization is to ensure that the most recently added schema comes first in the list. 

If an exceptional event type have say 50 schema versions for whatever reason, after some time, only the latest schema versions are likely to be used. So we don't want to compare with all the 49 before reaching the 50.

Reply:
I wonder why are we doing it in the first place...